### DISCUSSION TITLE
Mirth: Failed to migrate database schema

### DISCUSSION THREAD

**Discussion by roggss** [nextgenhealthcare/connect#5427] [Q&A]
HI,

MC 3.9.x

1. install MC
2. create new oracle user (with DBA grant, and all other rights copied from my working mirth user/schema)
3. configure the Database tab to oracle and URL
4. stop/start or restart the service

I checked the db session and MC session is there., it just did not create the tables.

error:
ERROR 2022-09-16 16:20:26,842 [Main Server Thread] com.mirth.connect.server.Mirth: Failed to migrate database schema
com.mirth.connect.model.util.MigrationException: java.sql.SQLSyntaxErrorException: ORA-00942: table or view does not exist​

Appreciate your help.

Thanks

09/23/2022 - follow up

Can somebody share the statements inside this file below (oracle-database.sql) or where to find it?
Thru this I can see what is goin on my database.

executeScript("/" + getDatabaseType() + "/" + getDatabaseType() + "-database.sql");

Note: I do have another instance of Mirth running on the same Oracle DB instance.

**Comment by pacmano1** [comment]
Does the named database for mirth exist?   e.g. "mirthdb"?   Not an Oracle user, but other DB engines other than Derby need the database to exist already.

**Comment by roggss** [comment]
Oracle don't have such or I maybe wrong. Can you point me on some links if you have?

Needless to say, I did the same thing with prior versions and it switched to Oracle with no issue.

Thanks

**Reply by pacmano1** [reply]
"Oracle don't have such or I maybe wrong".  What does that mean?

**Reply by roggss** [reply]
I mean, I don't know if there is a database naming for Oracle like other DB engines does.

**Reply by pacmano1** [reply]
I meant you must CREATE the database prior to starting mirth for other engines.

```SQL
create database mirthdb;
```
And whatever permission Oracle needs to allow your named user db owner permission on that DB.

**Reply by roggss** [reply]
Do you mean that I need to create a new database for Mirth alone? 
I believed, schema/user would suffice my needs, just like what I did on my other Mirth installations (old version)

**Reply by pacmano1** [reply]
Terminology difference, I see.  Yes, that should be fine.

**Reply by roggss** [reply]
I created mirthdb schema/user, still the same issue.

**Comment by roggss** [comment]
Hi,

Any other thoughts?

I am not sure if it is relevant, but I do have another instance of Mirth running on the same Oracle DB instance.

Regards

**Comment by roggss** [comment]
All,

The /conf/Oracle-database.sql is missing the script below for version 3.9.x Please do the needful. Thanks

create table D_CHANNELS
(
  local_channel_id NUMBER(19) not null,
  channel_id       NVARCHAR2(36) not null
)

**Reply by jonbartels** [reply]
Isn't that created via `createChannelsTable` in `oracle.xml`?

**Reply by roggss** [reply]
I don't see that file., or that snippet on xml files under conf/

**Reply by jonbartels** [reply]
https://github.com/nextgenhealthcare/connect/blob/442b0895ab988071a294b5fda68b59474a4de8b7/donkey/donkeydbconf/oracle.xml

**Reply by roggss** [reply]
Got it!

I only resolved my issue by manually creating all the tables (Oracle-database.sql + create table D_CHANNELS) after the fresh install.

Somehow, after fresh install it didn't create the oracle objects when switched from default db to Oracle.

Hope, you can give some insight on this.

Thanks
